privacy/ccpa/policy_test.go

```diff
@@ -539,43 +625,55 @@ func TestBuildExtWrite(t *testing.T) {
 		{
 			description:   "Leaves Other Ext.Prebid Values",
 			noSaleBidders: []string{"a", "b"},
-			ext:           json.RawMessage(`{"prebid":{"other":"any"}}`),
-			expected:      json.RawMessage(`{"prebid":{"nosale":["a","b"],"other":"any"}}`),
+			ext:           json.RawMessage(`{"prebid":{"aliases":{"a":"b"},"supportdeals":true}}`),
+			expected:      json.RawMessage(`{"prebid":{"aliases":{"a":"b"},"supportdeals":true,"nosale":["a","b"]}}`),
 		},
 		{
 			description:   "Leaves All Other Values",
 			noSaleBidders: []string{"a", "b"},
-			ext:           json.RawMessage(`{"other":"ABC","prebid":{"other":"123"}}`),
-			expected:      json.RawMessage(`{"other":"ABC","prebid":{"nosale":["a","b"],"other":"123"}}`),
+			ext:           json.RawMessage(`{"other":"ABC","prebid":{"aliases":{"a":"b"},"supportdeals":true}}`),
+			expected:      json.RawMessage(`{"other":"ABC","prebid":{"aliases":{"a":"b"},"supportdeals":true,"nosale":["a","b"]}}`),
 		},
 		{
 			description:   "Invalid Ext.Prebid No Sale Type - Still Overrides",
 			noSaleBidders: []string{"a", "b"},
 			ext:           json.RawMessage(`{"prebid":{"nosale":123}}`),
-			expected:      json.RawMessage(`{"prebid":{"nosale":["a","b"]}}`),
+			expected:      json.RawMessage(`{"prebid":{"nosale":123}}`),
+			expectedError: true,
```

Comment:
The tests in this file for malformed input have quite a different behavior. Before, it would ignore the bad input and write the proper input. Now, it throws an error because it needs to parse the input first. Would PBS have rejected the request for all of these cases? If so, these tests aren't really valid any more - we just need to leave one of them to test the error handling mechanism.

Reply:
That is a good question. Another question is would PBS even get this far, as wouldn't an unparseable blob within a larger JSON blob cause an error in parsing the larger blob? I don't think we can normally hit a parse error down here anyway.